pyracf/common/security_admin.py

```diff
@@ -157,35 +176,62 @@ def _make_request(
             self.__irrsmo00.call_racf(
-                security_request.dump_request_xml(), irrsmo00_precheck
+                security_request.dump_request_xml(),
+                irrsmo00_precheck,
+                self.__running_userid,
             ),
             self.__secret_traits,
         )
         self.__clear_state(security_request)
+        if isinstance(raw_result, list):
+            # When IRRSMO00 encounters some errors, it returns no XML response string.
+            # When this happens, the C code instead surfaces the return and reason
+            # codes which causes a DownstreamFatalError to be raised.
+            raise DownstreamFatalError(
+                saf_return_code=raw_result[0],
+                racf_return_code=raw_result[1],
+                racf_reason_code=raw_result[2],
+                request_xml=request_xml,
+                run_as_userid=self.get_running_userid(),
+            )
         if self.__debug:
             # No need to redact anything here since the raw result XML
             # already has secrets redacted when it is built.
-            self.__logger.log_xml("Result XML", result_xml)
-        results = SecurityResult(result_xml)
+            self.__logger.log_xml("Result XML", raw_result)
+        result = SecurityResult(raw_result, self.get_running_userid())
         if self.__debug:
             # No need to redact anything here since the result dictionary
             # already has secrets redacted when it is built.
             self.__logger.log_dictionary(
-                "Result Dictionary", results.get_result_dictionary()
+                "Result Dictionary", result.get_result_dictionary()
+            )
+        result_dictionary = result.get_result_dictionary()
+        if result_dictionary["securityResult"]["returnCode"] >= 8:
+            # All return codes greater than 4 are indicative of an issue with
+            # IRRSMO00 that would stop a command image from being generated.
+            # The user should interogate the result dictionary attached to the
+            # SecurityRequestError as well as the return and reason codes to
+            # resolve the problem.
+            raise DownstreamFatalError(
+                saf_return_code=8,
+                racf_return_code=result_dictionary["securityResult"]["returnCode"],
+                racf_reason_code=result_dictionary["securityResult"]["reasonCode"],
+                request_xml=request_xml,
+                run_as_userid=self.get_running_userid(),
+                result_dictionary=result_dictionary,
             )
-        result_dictionary = results.get_result_dictionary()
         if result_dictionary["securityResult"]["returnCode"] != 0:
             # All non-zero return codes should cause a SecurityRequestError to be raised.
-            # Even if a return code of 4 is not indicative of a problem, it it is
+            # Even if a return code of 4 is not indicative of a problem, it is
```

Comment:
the only listed return code 4 situations in IRRSMO00's doc is for both SAF and RACF RC 4. Theoretically this is true for either, and I think that specifying SAF could lead users to think RACF RC4 IS always indicative of a problem, whereas the more general statement covers both cases. I did tweak the wording and spacing a bit anyway.

Reply:
So in the comment should we say something about the SAF return code always matching the RACF return code?

Reply:
I mean in our dictionary it is just listed as a "returnCode" at this stage, and I think that adding the distinction in the comments makes that structure harder to follow.

Reply:
Good point. We can leave it as referring to `returnCode` in the result dictionary since talking about SAF and RACF might make things more confusing. I agree, keeping it simple and just referring to `returnCode` is better in that sense.